features-support/hooks/ui-automation.js

```diff
@@ -90,30 +91,41 @@ module.exports = function seleniumHooks() {
         .withCapabilities(capabilities)
         .build();
 
-      // Manage timeouts.
-      timeoutManager = driver.manage().timeouts();
-      timeoutManager.pageLoadTimeout(pageLoadTimeoutms);
-      timeoutManager.implicitlyWait(implicitlyWaitTimeoutms);
+      driver.getSession()
+          .then(function () {
+            // Manage timeouts.
+            timeoutManager = driver.manage().timeouts();
+            timeoutManager.pageLoadTimeout(pageLoadTimeoutms);
+            timeoutManager.implicitlyWait(implicitlyWaitTimeoutms);
 
+            driver.manage().window().maximize()
+                .then(function () {
+                  callback();
+                });
+
+          });
     } catch (error) {
```

Comment:
This is now a synchronous try-catch around an asynchronous operation. I think it's still needed to catch WebDriver instantiation issues but since the promise isn't returned it probably needs its own error handling as well?